ank/src/cli_commands.rs

```diff
@@ -223,42 +224,52 @@ impl CliCommands {
         let _ = self.task.await;
     }
 
+    async fn wait_for_complete_state(&mut self, object_field_mask: &Vec<String>, timeout_ms: u128) -> Result<Box<CompleteState>, CliError> {
+        output_debug!("wait_for_complete_state: object_field_mask={:?} timeout={timeout_ms} ", object_field_mask);
+
+        // send complete state request to server
+        self.to_server
+            .request_complete_state(RequestCompleteState {
+                request_id: self.cli_name.to_owned(),
+                field_mask: object_field_mask.clone(),
+            })
+            .await.map_err(|err| CliError::ExecutionError(err.to_string()))?;
+
+        let now = SystemTime::now();
+        loop {
```

Comment:
The loop will work and I do understand why you build in the timeout too. The thing is, it can still be that you never get a message and the timeout will never be triggered (endless wait).
I initially thought that we can endlessly wait as the response should come, but I do like your idea. The thing is it suggest that we will exit in 3 sec if nothing comes and this is now the case.

We could also wrap the `self.from_server.recv().await` in a `tokio::timeout`, but then the timeout will be kind of started again every time you receive an inappropriate message...

What do you think? Any better suggestion?

Reply:
Another suggestion would be to use `try_recv()` instead of `recv()` so that we can get rid of a possible 'endless wait'.

Reply:
With `try_recv()` did not work as expected as it returns always `TryRecvError::Empty`. Now I solved the problem with the 'endless wait' of `recv()` by using a `loop` within an `async` block that is wrapped in the `tokio::time::timeout`.